.github/workflows/new-release.yml

```diff
@@ -0,0 +1,66 @@
+name: New Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Version'
+        type: string
+        required: true
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+    - uses: actions/setup-node@v3
+      with:
+        node-version: 18
+        cache: 'yarn'
+    - name: Get version # TODO(eeeqeee) add option to increment from current version
+      id: get-version
+      run: |
+        echo "version=${{ inputs.version }}" >> $GITHUB_OUTPUT
+    - name: Set git config
+      run: |
+        git config user.name "github-actions[bot]"
+        git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
+    - name: Change version
+      run: |
+        TEMP_FILE=$(mktemp)
+        jq --arg v "${{ steps.get-version.outputs.version }}" '.version = $v' package.json > $TEMP_FILE
+        cp $TEMP_FILE package.json
+        git add .
+        git commit -m "${{ steps.get-version.outputs.version }}"
+    - name: Yarn install
+      run: |
+        yarn install --immutable --immutable-cache
+    - name: Setup .env
+      run: |
+        echo $ENVFILE > apps/frontend/.env.local
+      env:
+        ENVFILE: ${{ secrets.ENVFILE }}
+    - name: Build genshin-optimizer
+      run: |
+        NX_URL_GITHUB_GO_CURRENT_VERSION="https://github.com/${{ github.repository }}/commit/$(git rev-parse HEAD)"
+        yarn run nx run frontend:build:production
+    - name: Deploy to Github Pages
+      run: |
+        git fetch origin gh-pages --depth=1
+        git worktree add ../gh-pages gh-pages --no-checkout
+        cp -rT dist/apps/frontend ../gh-pages
+        cd ../gh-pages
+        git add --all .
+        git commit -m "deploy: version ${{ steps.get-version.outputs.version }} $(git rev-parse HEAD)"
+        git push
+    - name: Push
+      run: |
+        git status
+        git push
```

Comment:
This pushes the version change to the main repo, I am guessing? This step will ignore the `cd ../gh-pages` from the previous step and still be in the correct directory?

Reply:
yep, the current working directory should reset between steps